app/models/template.rb

```diff
@@ -84,13 +84,16 @@ def ignore_default
 
   # Set attributes based on template +text and metadata found in this +text
   # the metadata parsing can be adjusted using +options
-  def import_without_save(text, options = {})
+  def import_without_save(name, text, options = {})
     self.template = text
     @importing_metadata = self.class.parse_metadata(text)
-    Foreman::Logging.logger('app').debug "setting attributes for #{name} with id: #{id || 'N/A'}"
-    self.snippet = !!@importing_metadata[:snippet]
-    self.default = options[:default] unless options[:default].nil?
-    self.description = @importing_metadata[:description]
+    Foreman::Logging.logger('app').debug "setting attributes for #{name} with id: #{id || 'N/A'}" unless (options['associate'] == 'new')
+    if valid_params?(name, @importing_metadata)
+      self.snippet = !!@importing_metadata["snippet"]
+      self.default = options[:default] unless options[:default].nil?
+      self.description = @importing_metadata["description"]
+      self.template_kind_id = @importing_metadata["kind"]
```

Comment:
This is not correct per my assumption. Shouldn't metadata be a human readable name of the template kind and template_kind_id be an inner representation? This will not get translated automatically, we will need to look for the kind by its name.

Reply:
> Hi, the RM ticket nor the PR description explains what you're trying to fix here so it is hard to review. I'll assume that you want to import the kind from template file as `template_kind_id`.

Thanks for your comment. 
I updated the issue so it will be clearer.